aten/src/ATen/native/cudnn/MHA.cpp

```diff
@@ -553,7 +553,12 @@ void run_cudnn_SDP_fprop(
     Tensor& dropoutoffset) {
   cudnnHandle_t handle = getCudnnHandle();
   if (!o.defined()) {
-    o = at::empty({b, h, s_q, d_v}, q.options());
+    // q is passed to us in BHSD dim order
+    if (q.is_contiguous()) {
```

Comment:
So what I did for flex-attention is acutally just match the output striding to queries: 
https://github.com/pytorch/pytorch/pull/136905/files#diff-82826ad59d66d1141586434e8a67df782a0ee9e94020493af5f76f1d161f3e55R27

I think that in general this is the better option

Reply:
`empty_like` preserves stride information, right? Could use that